controllers/gitrepository_controller.go

```diff
@@ -450,6 +457,11 @@ func (r *GitRepositoryReconciler) reconcileSource(ctx context.Context,
 	defer cancel()
 	c, err := checkoutStrategy.Checkout(gitCtx, dir, repositoryURL, authOpts)
 	if err != nil {
+		var v libgit2.NoChangesError
+		if errors.As(err, &v) {
+			return sreconcile.ResultSuccess, nil
```

Comment:
@darkowlzz @hiddeco Do you see the need for a different status reporting here to highlight the `no-op`?

Reply:
I think no-op shouldn't report anything.

Reply:
btw, can't `errors.Is()` be used here?

Reply:
Looking at it more carefully, if we return success here, the following `reconcileArtifact()` will still expect `commit` variable to contain the commit information. It'd be better if we can populate the commit with the same information such that `reconcileArtifact()` doesn't find any diff and also becomes no-op.

Reply:
Thinking about the previous comment now:
> I think no-op shouldn't report anything.

When it's no-op, we can short-circuit the whole reconciliation by returning a typed error which is swallowed later at the end in `SummarizeAndPatch()` without resulting in an actual error. This may be better because if there's no source update and there's already an artifact in storage based on `reconcileStorage()`, we could skip `reconcileArtifact()`.

Reply:
I am now emitting an Waiting error instead, it does look and work a lot better. PTAL